New Resource/Data Source: `azurerm_nat_gateway` and `azurerm_subnet_nat_gateway_association` (#4449)
terraform-providers/terraform-provider-azurerm: website/docs/d/nat_gateway.html.markdown

> ---

Reviewer: This file will need to be added to the website index ERB

Author: Done

layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_nat_gateway"
sidebar_current: "docs-azurerm-datasource-nat-gateway"
description: |-
  Gets information about an existing Nat Gateway
---

# Data Source: azurerm_nat_gateway

Use this data source to access information about an existing Nat Gateway.

## Argument Reference

The following arguments are supported:

* `name` - (Required) Specifies the name of the Nat Gateway.

> * `resource_group_name` - (Required) Specifies the name of the Resource Group where the Nat Gateway exists.

Author: Changed

## Attributes Reference

The following attributes are exported:

* `location` - The location of the specified NAT Gateway.

* `idle_timeout_in_minutes` - The idle timeout of the specified Nat Gateway.

* `public_ip_address_ids` - A list of existing Public IP Address resource IDs which the specified NAT Gateway is using.

* `public_ip_prefix_ids` - A list of existing Public IP Prefix resource IDs which the specified NAT Gateway is using.

* `resource_guid` - The resource GUID property of the Nat Gateway.

* `sku_name` - The SKU used by the NAT Gateway.

* `subnet_ids` - A list of existing Subnet resource IDs.

* `zones` - A list of Availability Zones which the NAT Gateway should be created in .

> * `tags` - A mapping of tags to assign to the resource.

Author: Changed

---
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_nat_gateway"
sidebar_current: "docs-azurerm-datasource-nat-gateway"
description: |-
  Gets information about an existing Nat Gateway
---

# Data Source: azurerm_nat_gateway

Use this data source to access information about an existing Nat Gateway.

> 

Reviewer: extra white space here

> 

Reviewer: extra white space here

## Argument Reference

The following arguments are supported:

* `name` - (Required) Specifies the name of the Nat Gateway.

> * `resource_group_name` - (Required) Specifies the name of the Resource Group where the Nat Gateway should exist.

Reviewer: grammer
```suggestion
* `resource_group_name` - (Required) Specifies the name of the Resource Group where the Nat Gateway exists.
```

## Attributes Reference

The following attributes are exported:

> * `location` - Specifies the supported Azure location where the resource should exist.

Reviewer: grammer
```suggestion
* `location` - Specifies the supported Azure location where the resource exists.
```

> * `idle_timeout_in_minutes` - Specifies the idle timeout of the Nat Gateway.

Reviewer: This is worked like a resource, it should read like a data source

Author: I do not get this point. What should this be like for example?

> * `public_ip_address_ids` - A list of existing `azurerm_public_ip` resource IDs.

Reviewer: This is worked like a resource, it should read like a data source

> * `public_ip_prefix_ids` - A list of existing `azurerm_public_ip_prefix` resource IDs.

Reviewer: This is worked like a resource, it should read like a data source, what do these IDs mean

* `resource_guid` - The resource GUID property of the Nat Gateway.

> * `sku_name` - The Nat Gateway SKU.

Reviewer: ```suggestion
* `sku_name` - The SKU used by the NAT Gateway.
```

> * `subnet_ids` - A list of existing `azurerm_subnet` resource IDs.

Reviewer: This is worked like a resource, it should read like a data source. What are these IDs

> * `zones` - A list of availability zones where the Nat Gateway should be provisioned.

Reviewer: This is worked like a resource, it should read like a data source

---
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_nat_gateway"
sidebar_current: "docs-azurerm-datasource-nat-gateway"
description: |-
  Gets information about an existing Nat Gateway
---

# Data Source: azurerm_nat_gateway

Use this data source to access information about an existing Nat Gateway.

## Argument Reference

The following arguments are supported:

> * `name` - (Required) The name of the nat gateway.

Author: Done

> * `resource_group_name` - (Required) The Name of the Resource Group where the App Service exists.

Author: Done

## Attributes Reference

The following attributes are exported:

> * `location` - Resource location.

Author: Done

* `idle_timeout_in_minutes` - The idle timeout of the nat gateway.

* `public_ip_address_ids` - A list of IDs of existing `azurerm_public_ip` resources.

* `public_ip_prefix_ids` - A list of IDs of existing `azurerm_public_ip_prefix` resources.

* `resource_guid` - The resource GUID property of the NAT gateway resource.

* `sku_name` - The nat gateway SKU, supported values: Standard.

* `subnet_ids` - A list of IDs of existing `azurerm_subnet` resources.

> * `type` - Resource type.

Author: Removed
